### PR TITLE
10.76 [Close Please]

### PR DESCRIPTION
**Pull request by comedinha**
Enable login to 10.76

### CODE REVIEW

**Comment by marksamman** [comment]
Are you out of your mind?! Your pull request completely removes game world authentication. There is no way that is going to be merged, please just have the patience and let us do this properly when 1.1 is near-release.

**Comment by comedinha** [comment]
it was a bumpy way I found to get login on 10.74+, I'll see if I can find some other way

**Comment by comedinha** [comment]
The authentication removed to login in character, the first authentication to get the character list still exists.

**Comment by HeavenIsLost** [comment]
Mark can you teach us how to fix? At least we can have a alternative(fork) 10.75 tfs.

**Comment by DSpeichert** [comment]
@comedinha @HeavenIsLost I am not sure if you fully understand the outcome. Do you really want a fork for 10.75 that lets anybody to log in as any character without any password? Seems to me like a high price to pay for 10.75...

**Comment by comedinha** [comment]
In fact the protocolollogin.cpp makes user verification and password, I removed a second check.
The only problem we believe there is the character rename or is excluded with the open character list...

**Comment by Popzi** [comment]
So your saying that the removed checks from protocalgame is just a second safety check?

Else it would work for war servers without own accounts ;)

But to anyone with knowledge to fix this to a 100%, there is a $80 bounty on #887.  [Feel free to fix and claim it.](https://www.bountysource.com/issues/2998453-update-client-version-80)

**Comment by WibbenZ** [comment]
Items.xml isen't really needed, but it would be good.
Items.otb needs to be updated to fix animation phases added after 10.50~
Would be good if someone did this, I see this as a bigger issue then optimizing things at this point.

I get that Mark wants to keep his code for himself, but lets say 10.50-10.60 could be released since the twifysoft servers are using 10.70-10.75 right now.
Even the OX? server is using 10.76 and I get that they don't wanna release that version but letting everyone be at 10.41 isen't fun for everyone except those who can upgrade to a higher version.

**Comment by comedinha** [comment]
I will provide my otb 10.76, but I recommend not build it because I do not know how is the Mark otb.

https://mega.co.nz/#!W1sgEb6D!G_bVsGK75vOu9Uay2RBTitwCnI10IJknPCvoSOMPBBc

**Comment by WibbenZ** [comment]
Well that otb won't be accepted since the animation phases aren't added.
There are still bugs with ex. utana vid just to mark out one that is extrem.
But also the phase of ex. exevo gran mas frigo.

**Comment by HeavenIsLost** [comment]
if mean this:
	if (it.isAnimation) {
		AddByte(0xFE);    // random phase (0xFF for async)
	}

**Comment by comedinha** [comment]
Ty Cristofer Martins to help with while (std::getline(account, segment, '&'))

@WibbenZ Utana Vid and Exevo Gran Mas Frigo are normal for me...

**Comment by WibbenZ** [comment]
Nvm.. ill just show you:
Image
Hard to show shadowcores effect since it's way to quick.

The thing with mas frigo is what it's to slow - so takes longer till the effect is done.

**Comment by comedinha** [comment]
image
must be some new effect, I not have added effects

**Comment by comedinha** [comment]
I not have added frames = fin->getU8(); to otb
I will add later to see if he has anything to do

I'm trying to fix when the monster attacks you, is giving a strange debug

**Comment by luanlucian93** [comment]
perfect, you are brilliant guy!

File: src/protocolgame.cpp [hunk 1]
```diff
@@ -318,9 +318,17 @@ void ProtocolGame::onRecvFirstMessage(NetworkMessage& msg)
 	}
 
 	msg.SkipBytes(1); // gamemaster flag
-	std::string accountName = msg.GetString();
+	std::string sessionKey = msg.GetString();
+	std::stringstream account(sessionKey);
+	std::string segment;
+	std::vector<std::string> seglist;
+	while (std::getline(account, segment, '&'))
+	{
```
Comment (WibbenZ) [inline src/protocolgame.cpp]: Should be on line 326 ) {

**Comment by comedinha** [comment]
I fix some packet erros, in otclient everything is perfect, but in the normal client bug occurs when a monster attacks... That is, any packet that there is in otclient

**Comment by WibbenZ** [comment]
One thing I noticed is the expert mode setting button is as I suppose it is on retro open pvp servers / hardcore servers.
http://tibia.wikia.com/wiki/Updates/10.73
"The PvP Expert Mode in characters' combat control is not visible anymore on Hardcore and Retro Open PvP game worlds."

**Comment by WibbenZ** [comment]
ProtocolGame::ParseServerPvpSituations
Should have an AddByte I think not GetByte.

Same with ProtocolGame::ParseServerUnjustifiedPoints since it returns things about your skull and skull time etc.

And the combat issue works for players aswell as monsters(damage taker gets client crash)

**Comment by HeavenIsLost** [comment]
We are fixing that

**Comment by comedinha** [comment]
10.76 attacks fixed
Removed the other packets, if someone wants to work on them...

**Comment by uMapinha** [comment]
LOOOL
awesome!!!!

**Comment by WibbenZ** [comment]
Great! :)
But do you know what needs to be changed to open expert mode?
Trying to fix that system and worked on 10.55 that I used before.

**Comment by comedinha** [comment]
@WibbenZ Added the expert mode by settings, yet I doubt if I put for check the pvp type instead config

**Comment by WibbenZ** [comment]
Thanks! Gonna recompile and see if it works.
How ever that config boolean is not needed, since IF we are going to change the pvp system it depends on the world pvp setting.
Hardcore or retro PVP  = disabled, otherwise enabled.

But would be good if someone wants to disable them when the new warsystem is added - if it ever gets added.

**Comment by comedinha** [comment]
I switched to check the type of pvp, if you have this feature (PVP Mode) already made would be a good :)

**Comment by WibbenZ** [comment]
Well if people do things like this ill make sure to either upload it or make someone upload it (my github client dosen't like me :p)
Right now ive just been playing around abit with it, I use lua functions and the onTargetCombat event to execute the code.
Tought about doing it in C++ to not have to add new lua functions that most likely won't be used any other time...But I agree with Mark that we should move as much as posible to lua.
And the config boolean you added can also be used in this case to check if it should ignore pvp mode or not.

**Comment by Scrux** [comment]
And can run on tfs 1.0?

**Comment by comedinha** [comment]
@Schiffers Yes, last tfs 1.1

@Scrux I know some people who have succeeded, others are having some problems...

**Comment by gpedro** [comment]
@comedinha you're fucking awesome, man!

**Comment by Scrux** [comment]
does not work otb 10.76, map 10.7, debug when it comes

**Comment by comedinha** [comment]
otb 10.76
https://mega.co.nz/#!fsFGGCrS!7y0Tp_Q-nFImloDF78TxkjbEXI-Bl_m_Kfi7ItYj63s

Map editor 10.76
https://mega.co.nz/#!e91RXYQC!k6GzJYNch_W9muI7wuaOp050WtQkcwrYw4UG3rN72Lo

This is not an official otb, use at your own risk. Recommend not start working on these ids...

**Comment by Scrux** [comment]
continues the debug must be something else

**Comment by comedinha** [comment]
I do not know, my server and map are equal to everything I said here...

image

**Comment by WibbenZ** [comment]
@comedinha You got any ides on the unjustified tab yet? :p

**Comment by Scrux** [comment]
I'm using tfs 1.0

**Comment by comedinha** [comment]
@WibbenZ No, tfs only works with daily frags... I tried adding at least one part and leave the weekly and monthly aside but got strange problems

```
void ProtocolGame::AddUnjustifiedStats(NetworkMessage& msg, const Creature* creature)
{
 if (creature != player) {
  return;
 }

 msg.AddByte(player->getSkullTicks() / g_config.getNumber(ConfigManager::FRAG_TIME));
 msg.AddByte(g_config.getNumber(ConfigManager::KILLS_TO_RED));
 msg.AddByte(0x00);
 msg.AddByte(0x00);
 msg.AddByte(0x00);
 msg.AddByte(0x00);
 msg.AddByte((player->getSkullTicks() % g_config.getNumber(ConfigManager::FRAG_TIME)) / 1000);
}
```

**Comment by WibbenZ** [comment]
Well the most importent thing would be the frag count etc.
Got a hard time understanding that thingi, haven't play rl after they added it.

But the good thing if it gets added is that we can remove the !kills command.

**Comment by mimusassasin** [comment]
hey guys! all of u are awsome!!!! like we say in mexico "son una verga!" jeje well first tank u so much this is working perfect for me on the printerlua data... but i have a little bug i tink 'cause znote acc's playerdeath.lua (for the guildwar stuff) well the error is this


Lua Script Error: [CreatureScript Interface] 
data/creaturescripts/scripts/others/playerdeath.lua:onDeath
data/creaturescripts/scripts/others/playerdeath.lua:47: attempt to concatenate local 'mostDamage_unjustified' (a boolean value)
stack traceback:
	[C]: in function '__concat'
	data/creaturescripts/scripts/others/playerdeath.lua:47: in function <data/creaturescripts/scripts/others/playerdeath.lua:4>

im trying to lerning all i can but this is over me T.T any can explain what this error means?

**Comment by Codinablack** [comment]
@mimusassasin https://otland.net/forums/support.16/

**Comment by absoluten** [comment]
here is Braaaaaaaaaaazil!
good Comédia, you are out of this world. ;)
#MARKESTABOLADAO

**Comment by Schiffers** [comment]
Hey Comedinha you is a nice guy :D 
I have one question, bro, have debug when I put all mounts ? know, the client support only 50 mounts so,
you have fixed this? for less to see the first 50 mounts, if I log with an account gamemaster ....

**Comment by renatorib** [comment]
#MARKESTABOLADAO
HHAHAAAHUAHAHHAUHAUHAUA

**Comment by renatorib** [comment]
PS: huebr

**Comment by marksamman** [comment]
Stop spamming please, your messages do not contribute anything useful to this pull request.

**Comment by Nu77** [comment]
What is this? Why this stupid behavior, seriously... If you want so much that feature, only try do it, try and try again. Don't be a stupid without reason.

**Comment by renatorib** [comment]
@gpedro @comedinha @absoluten @luanlucian93 to começando a achar que não é br que é zoero, mas os gringos que são muito chatos.

@marksamman sorry for spamming. I just think his first comment was extremely unnecessary. Here the translation of "Are you out of your mind?" gets a little offensive. Comedinha is just trying to contribute and you did so: "stop being stupid, we're the best, let we solve this". Pathetic

**Comment by WibbenZ** [comment]
I can agree that I also reacted on Marks comment, but I still get he's reaction.
Also went abit :o when I read the changes he made.
But that isen't a reason for spamming, I would be complaining if he closed it. But he diden't.

**Comment by absoluten** [comment]
@renatorib @luanlucian93 @gpedro @comedinha meça suas palavras parça

@marksamman I'm sorry, but I take liberty to interfere in the case.
"Are you out of your mind?" do not need that right mate?
Anyway, I'm waiting for the update of you developers TFS in this protocol.
Hugs.

**Comment by marksamman** [comment]
Unfortunately, the pull request had to be locked because of spam from certain individuals who could not resist to show their immaturity to the world. I'm sorry that we can't collaborate any more on this pull request because of the people this pull request has attracted, @comedinha. 

As for technical matters, there are still multiple issues with the patch; game world authentication does work now (it was never just a second check; it is the only check in game protocol, it is critical for security, hence why I harshly questioned your decision to remove it), but has not been implemented properly and in an efficient manner. I will be pushing protocol 10.76 support with the official OTB eventually as we are closing in on the 1.1 release, so this pull request is not going to be merged but I will leave it open to so that it's easier to discover because of the demand for 10.76, keep in mind that you may run into a lot of issues with it though.

Thank you @Popzi for placing a bounty on the issue (#887), it's the most meaningful contribution among all the crap I've seen around this issue thus far.

**Pull request closed without merging**